### PR TITLE
ramips: add support for D-Link DIR-1935 A1

### PR DESCRIPTION
**Pull request by kewiha**
Add support for D-Link DIR-1935 A1 based on similarities to DIR-882 A1, DIR-867 A1 and other DIR-8xx A1 models. Existing DIR-882 A1 openwrt "factory" firmware installs without modificaitons via the D-Link Recovery GUI and has no known incompatibilities with the DIR-1935 A1.

Changes to be committed:
       new file:   target/linux/ramips/dts/mt7621_dlink_dir-1935-a1.dts
       modified:   target/linux/ramips/image/mt7621.mk
       modified:   target/linux/ramips/mt7621/base-files/etc/board.d/01_leds

Specifications:
* Board: Not known
* SoC: MediaTek MT7621 Family
* RAM: 128 MB (DDR3)
* Flash: 16 MB (SPI NOR)
* WiFi: MediaTek MT7615 Family (x2)
* Switch: 1 WAN, 4 LAN (Gigabit)
* Ports: 1 USB 3.0
* Buttons: Reset, WiFi Toggle, WPS
* LEDs: Power (green/orange), Internet (green/orange), WiFi 2.4G (green), WiFi 5G (green)

Notes:
* 160MHz 5GHz is available in LuCi but does not appear to work (i.e. no SSID is visible in wifi scanning apps on other devices) with either official DIR-882 A1 firmware or a test build for the DIR-1935 A1 based on the 22.03.2 branch. 80 MHz 5GHz works.

Serial port:
* Untested (potential user damage/error)
* Expected to be identical to other DIR-8xx A1 models:
	* Parameters: 57600, 8N1
	* Location: J1 header (close to the Reset, WiFi and WPS buttons)
	* Pinout: 1 - VCC 2 - RXD 3 - TXD 4 - GND

Installation:
* D-Link Recovery GUI: power down the router, press and hold the reset button, then re-plug it. Keep the reset button pressed until the power LED starts flashing orange, manually assign a static IP address under the 192.168.0.xxx subnet (e.g. 192.168.0.2) and go to http://192.168.0.1

* Some modern browsers may have problems flashing via the Recovery GUI, if that occurs consider uploading the firmware through cURL:

    curl -v -i -F "firmware=@file.bin" 192.168.0.1

Signed-off-by: Keith Harrison <keithh@protonmail.com>

### CODE REVIEW

**Comment by kewiha** [comment]
Associated forum thread: https://forum.openwrt.org/t/dir-1935-supported/145614/1

File: target/linux/ramips/mt7621/base-files/etc/board.d/01_leds [hunk 1]
```diff
@@ -212,3 +213,4 @@ esac
 board_config_flush
 
 exit 0
+
```
Comment (hauke) [inline target/linux/ramips/mt7621/base-files/etc/board.d/01_leds:216]:
```suggestion
```
Please do not add this unrelated change.

Comment (kewiha) [reply target/linux/ramips/mt7621/base-files/etc/board.d/01_leds]: Thank you

**Comment by DragonBluep** [comment]
> * 160MHz 5GHz is available in LuCi but does not appear to work (i.e. no SSID is visible in wifi scanning apps on other devices) with either official DIR-882 A1 firmware or a test build for the DIR-1935 A1 based on the 22.03.2 branch. 80 MHz 5GHz works.

DIR-1935 is a AC1900 device (It should uses [MT7615S](https://www.mediatek.com/products/broadband-wifi/mt7615s) 3x3 in 2GHz and [MT7615B](https://www.mediatek.com/products/broadband-wifi/mt7615b) 3x3 in 5 GHz), so the 160MHz support report is incorrect. It seems that the `mt76` driver does not support these 3x3 devices very well. Anyway, it doesn't support 160MHz, should be same as DIR-876

And you can squash the second commit into the first one.

**Comment by kewiha** [comment]
> > * 160MHz 5GHz is available in LuCi but does not appear to work (i.e. no SSID is visible in wifi scanning apps on other devices) with either official DIR-882 A1 firmware or a test build for the DIR-1935 A1 based on the 22.03.2 branch. 80 MHz 5GHz works.
> 
> DIR-1935 is a AC1900 device (It should uses [MT7615S](https://www.mediatek.com/products/broadband-wifi/mt7615s) 3x3 in 2GHz and [MT7615B](https://www.mediatek.com/products/broadband-wifi/mt7615b) 3x3 in 5 GHz), so the 160MHz support report is incorrect. It seems that the `mt76` driver does not support these 3x3 devices very well. Anyway, it doesn't support 160MHz, should be same as DIR-867.
> 
> And you can squash the second commit into the first one.

Squashed it. Also added a note about a potential bug in the 2.4GHz wifi to the commit message (80MHz not listed, should be available based on the D-link [datasheet](https://ftp.dlink.ca/ftp/PRODUCTS/DIR-1935/DIR-1935_REVA_DATASHEET_1.00_EN.PDF)). 

Thanks

**Comment by DragonBluep** [comment]
> Squashed it. Also added a note about a potential bug in the 2.4GHz wifi to the commit message (80MHz not listed, should be available based on the D-link [datasheet](https://ftp.dlink.ca/ftp/PRODUCTS/DIR-1935/DIR-1935_REVA_DATASHEET_1.00_EN.PDF)). 

It's not a bug. 802.11n (2.4 GHz) only support 20/40 MHz BW. That's why 5 GHz is faster than 2.4 GHz.
As written in the datasheet, the maximum speed is 1300 at 5GHz and 600 at 2.4GHz, 1300/600≈80MHz/40MHz.

> Should I make an issue (or two?) in the mt76 repo about the wifi band issues?

If you are really bothered by this 160MHz problem, perhaps you can open an issue in [mt76](https://github.com/openwrt/mt76) and ask the developer to remove the 160MHz support of MT7615B.

**Comment by kewiha** [comment]
> It's not a bug. 802.11n (2.4 GHz) only support 20/40 MHz BW. That's why 5 GHz is faster than 2.4 GHz. As written in the datasheet, the maximum speed is 1300 at 5GHz and 600 at 2.4GHz, 1300/600≈80MHz/40MHz.

Revised my commit message to not mention 2.4GHz. Thank you

**Review by DragonBluep — APPROVED** [review]
